geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/ListGatewayCommand.java

```diff
@@ -53,9 +53,51 @@ public ResultModel listGateway(
           help = CliStrings.LIST_GATEWAY__MEMBER__HELP) String[] onMember,
       @CliOption(key = {CliStrings.GROUP, CliStrings.GROUPS},
           optionContext = ConverterHint.MEMBERGROUP,
-          help = CliStrings.LIST_GATEWAY__GROUP__HELP) String[] onGroup)
+          help = CliStrings.LIST_GATEWAY__GROUP__HELP) String[] onGroup,
+      @CliOption(key = {CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY},
+          specifiedDefaultValue = CliStrings.LIST_GATEWAY__SHOW_PARAM_TRUE,
+          unspecifiedDefaultValue = CliStrings.LIST_GATEWAY__SHOW_PARAM_EMPTY,
+          help = CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY) String showReceiversOnly,
+      @CliOption(key = {CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY},
+          specifiedDefaultValue = CliStrings.LIST_GATEWAY__SHOW_PARAM_TRUE,
+          unspecifiedDefaultValue = CliStrings.LIST_GATEWAY__SHOW_PARAM_EMPTY,
+          help = CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY__HELP) String showSendersOnly)
+
       throws Exception {
 
+    // Check value of --receivers-only
+    if (!showReceiversOnly.equals(CliStrings.LIST_GATEWAY__SHOW_PARAM_EMPTY)
+        && !showReceiversOnly.equals(CliStrings.LIST_GATEWAY__SHOW_PARAM_TRUE)
+        && !showReceiversOnly.equals(CliStrings.LIST_GATEWAY__SHOW_PARAM_FALSE)) {
+      return ResultModel.createError(CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY_INPUT_ERROR);
+    }
+
+    // Check value of --senders-only
+    if (!showSendersOnly.equals(CliStrings.LIST_GATEWAY__SHOW_PARAM_EMPTY)
+        && !showSendersOnly.equals(CliStrings.LIST_GATEWAY__SHOW_PARAM_TRUE)
+        && !showSendersOnly.equals(CliStrings.LIST_GATEWAY__SHOW_PARAM_FALSE)) {
+      return ResultModel.createError(CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY_INPUT_ERROR);
+    }
+
+    // Check --senders-only & --receivers-only have not been used together
+    if (!showReceiversOnly.equals(CliStrings.LIST_GATEWAY__SHOW_PARAM_EMPTY)
+        && !showSendersOnly.equals(CliStrings.LIST_GATEWAY__SHOW_PARAM_EMPTY)) {
+      return ResultModel.createError(CliStrings.LIST_GATEWAY__ERROR_ON_SHOW_PARAMETERS);
+    }
```

Comment:
Can you move this logic to an interceptor -> `@CliMetaData(interceptor = "interceptor.class.name")`?, it's the standard way of doing this kind of input validation for GFSH commands. You can have a look at `AlterAsyncEventQueueCommand` and `CreateRegionCommand` for examples about its usage.